### ISSUE TITLE
[2.4.0][Watch] Renaming a file causes parser errors

### ISSUE THREAD

**Issue by fengcen** [typescript-eslint/typescript-eslint#1081]
Rename file to subdirectory will throw this error.

**What code were you trying to parse?**

```ts
// Please put code here
```

**What did you expect to happen?**
Do not throw error

**What actually happened?**
```
Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: [refactor-file-path].
The file must be included in at least one of the projects provided.
```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.4.0` |
| `TypeScript`                | `3.6.4` |
| `ESLint`                    | `6.5.1` |
| `node`                      | `12.10.0` |
| `npm`                       | `6.11.2` |

**Comment by bradzacher** [comment]
Refactor what? What steps did you take to trigger the error?
Could you please provide some more information?

**Comment by fengcen** [comment]
> Refactor what? What steps did you take to trigger the error?
> Could you please provide some more information?

Rename file to other directory will throw this error.

**Comment by bradzacher** [comment]
There is an unavoidable race condition between the IDE opening the file, and typescript picking up the change, so certain actions (creating a new file and opening it instantly, moving an open file to a new directory, etc) will trigger this.

Closing and reopening the file should fix it.

**Comment by fengcen** [comment]
No. I try to reopening the file and the error still there.

**Comment by bradzacher** [comment]
Hmm, it appears that moving a file triggers a different set of file system events than creating a file.

**Comment by bradzacher** [comment]
Closing in favour of #1091
